### ISSUE TITLE
distDir containing trailing slash

### ISSUE THREAD

**Issue by yuyichao** [zeit/next.js#10582]
# Bug report

## Describe the bug

This issue may have happened after upgrading from node 13.3.0 (or 13.2.0) to 13.7.0 recently. After the upgrade, the test server suddenly stops working and is giving 404 on a lot of js file requests. After a while I noticed that my `static/deveopment/pages` contains a subdirectory that I don't remember seeing.

```
static/development/pages
├── _app.js
├── _app.js.map
├── _error.js
├── _error.js.map
├── index.js
├── index.js.map
└── next
    └── dist
        └── pages
            ├── _error.js
            └── _error.js.map
```

which may or may not be normal but that lead me to check my `distDir` setting and I noticed that I was setting a `distDir` in my `next.config.js` to a path that has a trailing `/`. If that slash is removed then the issue is gone. If I set it to `.next/`, then it stops working so the trailing slash, rather than any other part of my `distDir`, seems to be the issue.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. `git clone https://github.com/zeit/next-learn-demo`
2. `cd next-learn-demo/4-clean-urls`
3. `npm update`
4. `node_modules/next/dist/bin/next` and check `localhost:3000` in browser, this should work.
5. Create a `next.config.js` with content

    ```
    module.exports = {
        distDir: '.next/',
    };
    ```
6. `node_modules/next/dist/bin/next` and now `localhost:3000` in browser gives a blank page with a lot of 404 in the console... No error what so ever is printed in the terminal...

## Expected behavior

Should still work with trailing slash. It used to work (80% sure, though the `distDir` change was one of the last things I did before stop working on it for a while so I may not have checked too carefully....) If there's good reason to disallow this, it should be mentioned in [the doc](https://nextjs.org/docs/api-reference/next.config.js/setting-a-custom-build-directory)

## Screenshots

N.A.

## System information

- OS: ArchLinux
- Browser (if applies) Chromium 80.0.3987.87 Arch Linux
- Version of Next.js: `9.2.2`

## Additional context

N.A.

**Comment by fabianishere** [comment]
I have narrowed the issue down to `isServeableUrl` in `next-server.ts`:
https://github.com/zeit/next.js/blob/082cc68d87854034e95db371af32717d83bad2f6/packages/next/next-server/server/next-server.ts#L1260-L1272

In particular, the fragment `join(this.distDir) + sep` results in a path with two separators at the end, since `this.distDir` still contains the trailing slash. Would an appropriate solution be to join the path separator instead? That is:

```js
join(this.distDir, sep)
```

**Comment by timneutkens** [comment]
This has been resolved in 9.3.2

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
